variant-caller-wdl/topmed_freeze8_caller.wdl

```diff
@@ -0,0 +1,811 @@
+version 1.0
+
+## This is the U of Michigan variant caller workflow WDL for the workflow code located here:
+## https://github.com/statgen/topmed_variant_calling
+##
+## It uses a Docker image built with software tools that can reproduce
+## variant calls compatible to TopMed Freeze 8
+##
+## NOTE: This workflow assumes that input CRAM files have been built with the b38
+## human reference genome. In particular for the TopMed CRAM files the
+## reference genome files to use are located here:
+## ftp://share.sph.umich.edu/1000genomes/fullProject/hg38_resources
+## wget ftp://share.sph.umich.edu/1000genomes/fullProject/hg38_resources/topmed_variant_calling_example_resources.tar.gz
+##
+##
+
+workflow TopMedVariantCaller {
+    input {
+      Boolean dynamically_calculate_disk_requirement = true
+
+      Float All_CRAMs_disk_size_override = 1000.0
+      Float All_CRAIs_disk_size_override = 100.0
+      Float CRAM_file_max_disk_size_override = 200.0
+      Float ReferenceGenome_disk_size_override= 30.0
+
+      Int SumFileSizes_preemptible_tries = 3
+      Int SumFileSizes_maxretries_tries = 3
+      Int SumFileSizes_memory = 7
+      Int SumFileSizes_CPUs = 1
+      Int SumFileSizes_disk_size = 1
+
+      Int CreateCRAMIndex_preemptible_tries = 3
+      Int CreateCRAMIndex_maxretries_tries = 3
+      Int CreateCRAMIndex_memory = 7
+      Int CreateCRAMIndex_CPUs = 1
+
+      # The variant caller could take more than 24 hours to run. GCP terminates
+      #  preemptible tasks after 24 hours. So by using 0 for preemptible tries the
+      #  task is non preemtible
+      #  if preemptible is set to 0 -- then its set to false
+      #  if preemptible is set to a positive integer -- its automatically true
+      Int VariantCaller_preemptible_tries = 1
+      #if preemptible is 0 and maxRetries is 3 -- then that task can be retried upto 3 times
+      #if preemptible is 3 and maxRetries is 3 for a task -- that can be retried upto 6 times
+      #https://cromwell.readthedocs.io/en/stable/RuntimeAttributes/#maxretries
+      Int VariantCaller_maxretries_tries = 3
+      # Select memory and CPUs to choose a GCP n1-highmem-64 machine
+      Int VariantCaller_CPUs = 64
+      Int VariantCaller_memory = 400
+      # For adding more disk space for the variant caller from an input file
+      Int VariantCaller_additional_disk = 1
+
+      String variantCallerHomePath =  "/topmed_variant_calling"
+
+      File referenceFilesBlob
+      Array[File] referenceGenomeFilesTarGz = [referenceFilesBlob]
+
+      Array[File]? input_crai_files
+      Array[File] input_cram_files
+      Array[String] input_cram_files_names = input_cram_files
+
+      String docker_image = "statgen/topmed-variant-calling:v8.0.4"
```

Comment:
I don't write workflows, so I may be wrong, but I think we should move away from allowing the docker image to be configurable.

Reply:
I will leave the Docker image as configurable for now since it has been so helpful in the past when I noticed missing tools and bugs in the Dockerfile where the original authors provided a fix in a new Docker image release. A researcher should include the input file they used to produce data along with the workflow code; and that would help provide documentation on the Docker image that was used.